### PR TITLE
RELATED: SD-2301 Fixed dropdown help menu wrong position

### PR DESCRIPTION
**Pull request by dlcd2203**
JIRA: SD-2301

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [x] `check` passes
-   [x] `check-extended` passes
-   [x] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by dlcd2203** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8911/ : SUCCESS in 15s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10760/ : SUCCESS in 15m 22s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1730/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8912/ : SUCCESS in 15s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1730/ : SUCCESS in 27m 17s

**Comment by dlcd2203** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8913/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10761/ : SUCCESS in 15m 20s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1731/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8914/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1731/ : SUCCESS in 26m 21s

**Comment by dlcd2203** [comment]
extended check cypress

**Comment by yenkins** [comment]
# ✨ Cypress test results ✨

## 🎉 Great news! Tests were successful.

|Status|Name|Tests|Failures|Time|
|:--:|:--|:--:|:--:|--:|
|✅|**pivotTableRemoveOneFromMutipleAggregation.spec.ts**|1|-|11.3590s|
|✅|**pivotTableRemoveAllFromMutipleAggregation.spec.ts**|1|-|11.1100s|
|✅|**dashboardFiltering.spec.ts**|2|-|24.4020s|
|✅|**headlineChartDrilling.spec.ts**|2|-|8.0650s|
|✅|**pivotTableAggregationsMenuAndAddTotal.spec.ts**|5|-|23.8900s|
|✅|**attributeFilterButtonParentChild.spec.ts**|3|-|18.3120s|
|✅|**pivotTableRemoveOneAggregation.spec.ts**|1|-|11.1400s|
|✅|**drilling.spec.ts**|2|-|20.1530s|
|✅|**attributeFilterButtonWithSelection.spec.ts**|2|-|10.7330s|
|✅|**attributeFilterConfig.spec.ts**|3|-|29.4350s|
|✅|**pivotTableSizingReset.spec.ts**|15|-|42.3380s|
|✅|**dateFilter.spec.ts**|9|-|17.5580s|
|✅|**attributeFilterButton.spec.ts**|9|-|34.6570s|
|✅|**pivotTableSizing.spec.ts**|1|-|9.0680s|
|✅|**parentChildFiltersWithPlaceholders.spec.ts**|3|-|22.3550s|
|✅|**barChartDrilling.spec.ts**|4|-|10.7630s|
|✅|**dashboard.spec.ts**|9|-|53.2190s|

You can find full report [here](https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/257/).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8915/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/257/ : SUCCESS in 13m 50s

File: libs/sdk-ui-kit/src/Header/HeaderHelp.tsx [hunk 1]
```diff
@@ -128,7 +133,7 @@ export const CoreHeaderHelp: React.FC<IHeaderHelpProps> = ({
             <FormattedMessage id="gs.header.help" />
         </a>
     ) : (
-        <div className={classNames} onClick={() => toggleHelpMenu()}>
+        <div className={classNames} onClick={() => toggleHelpMenu()} ref={helpMenuRef}>
```
Comment (anvuong119) [inline libs/sdk-ui-kit/src/Header/HeaderHelp.tsx:136]: Can we simply pass `toggleHelpMenu` here instead, something like:

```
<div className={classNames} onClick={() => toggleHelpMenu()} ref={helpMenuRef}>
```

↓

```
<div className={classNames} onClick={toggleHelpMenu} ref={helpMenuRef}>
```

Comment (dlcd2203) [reply libs/sdk-ui-kit/src/Header/HeaderHelp.tsx:136]: Screen Shot 2022-07-26 at 15 00 54
We can't, it gets error tslint.

Comment (anvuong119) [reply libs/sdk-ui-kit/src/Header/HeaderHelp.tsx:136]: ok then

**Comment by anvuong119** [comment]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2785

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2551/ : SUCCESS in 1m 09s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2850/ : SUCCESS in 20s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1679/ : SUCCESS in 19m 19s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10764/ : SUCCESS in 15m 13s

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2785

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Pull request merged**